tests/ubpf_vm.rs

```diff
@@ -2227,6 +2228,19 @@ fn test_vm_stxw() {
     assert_eq!(vm.execute_program(mem).unwrap(), 0x44332211);
 }
 
+#[test]
+fn test_bpf_to_bpf_call(){
+    let mut program = BpfCode::new();
+    program.mov(Source::Imm, Arch::X64).set_dst(0x1).set_imm(0xff).push()
+    .call().set_src(1).set_imm(1).push()
+    .mov(Source::Imm, Arch::X64).set_dst(0x2).set_imm(0x1).push()
+    .mov(Source::Imm, Arch::X64).set_dst(0).set_imm(0xf).push()
+    .exit().push();
+    let prog = program.into_bytes();
+    let vm = rbpf::EbpfVmNoData::new(Some(&prog)).unwrap();
+    assert_eq!(vm.execute_program().unwrap(), 0xf);
+}
+
```

Comment:
The assembly syntax would be more intuitive to read. Did you fall back on the chained API because the assembler doesn't support function calls?